### ISSUE TITLE
Improve shape mismatch error in layers.

### ISSUE THREAD

**Issue by nsthorat** [tensorflow/tfjs#1165]
When calling fit, we get errors like this:

```
tf-layers.esm.js:17 Uncaught (in promise) Error: Error when checking target: expected sampling_layer_SamplingLayer1 to have shape [,4], but got array with shape [1,512].
```

This is mostly fine, but it would be good to log the name of the other layer that this connects with that is causing the error.

Maybe also instead of "target" we could say something like "output"? How about this error message:

```
sampling_layer_SamplingLayer1 is expected to have an input shape of [null, 4] but the output of previous_layer has an output shape of [null, 512].
```

**Comment by usmanmuhd** [comment]
Hey! Interested in this. Any starting point that I should be looking at?

**Comment by duhaime** [comment]
@caisq are you working on this one? If not, is it okay for the community to send a PR?

**Comment by caisq** [comment]
@duhaime @usmanmuhd Go for it.

**Comment by sainikhit2k** [comment]
Hi , is anybody working on this issue ? If not, I would like to work on this issue

**Comment by duhaime** [comment]
I had hoped to but am bound up at the moment!

**Comment by dhirensr** [comment]
@nsthorat @duhaime : i have sent a PR for this issue https://github.com/tensorflow/tfjs/pull/2848 Please review and let me know if there are any changes required. :smile:

**Comment by Nikki-2809** [comment]
I will work on this.

**Comment by sourabh112** [comment]
Hi, is anybody working on this issue? If not, I would like to work on this issue.

**Comment by rthadur** [comment]
@sourabh112 please go ahead with the changes , thank you for your interest.

**Comment by mdaj06** [comment]
@sourabh112 if you are stuck somewhere could i help? Or if you are busy at the moment can i pick it up?

**Comment by sourabh112** [comment]
@mdaj06 Thanks for tagging me. My semester has started and I totally missed updates this issue. I'm busy right now and can't take on this issue for now, you can go ahead and work on it.

**Comment by mdaj06** [comment]
@sourabh112 sure no problem! @rthadur i'll have a go at this issue then

**Comment by pree-T** [comment]
@mdaj06 I want to work on this. Will you please help me as I am a beginner.

**Comment by Gua00va** [comment]
Is this issue still open? I would like to work on this.

**Comment by prakhart4** [comment]
As the current error at [training.ts](/tensorflow/tfjs/blob/master/tfjs-layers/src/engine/training.ts#L170-L179) looks like this:

~~~ts
throw new ValueError(
              `${exceptionPrefix} expected a batch of elements where each ` +
              `example has shape [${shapes[i].slice(1, shapes[i].length)}] ` +
              `(i.e.,tensor shape [*,${
                  shapes[i].slice(1, shapes[i].length)}])` +
              ` but the ${exceptionPrefix} received an input with ${
                  array.shape[0]}` +
              ` examples, each with shape [${
                  array.shape.slice(1, array.shape.length)}]` +
              ` (tensor shape [${array.shape}])`);
        }
~~~

The issue can be considered resolved.

**Comment by gaikwadrahul8** [comment]
Hi, @dhirensr 

Apologize for the delayed response and just to confirm, May I know have you re-submitted the PR with email which you signed CLA please or have we taken care of this issue with latest version `@tensorflow/tfjs@4.10.0` ? Thank you!

**Comment by rubiagatra** [comment]
I also take a look for this one @gaikwadrahul8

**Comment by rubiagatra** [comment]
I think this is issue solved @gaikwadrahul8 already fixed on `@tensorflow/tfjs@4.10.0` https://github.com/tensorflow/tfjs/blob/tfjs-v4.10.0/tfjs-layers/src/engine/training.ts#L170
